python/gstaichi/lang/kernel_impl.py

```diff
@@ -1102,10 +1116,13 @@ def ast_to_dict(node: ast.AST | list | primitive_types._python_primitive_types):
     def launch_kernel(self, t_kernel: KernelCxx, compiled_kernel_data: CompiledKernelData | None, *args) -> Any:
         assert len(args) == len(self.arg_metas), f"{len(self.arg_metas)} arguments needed but {len(args)} provided"
 
-        actual_argument_slot = 0
-        launch_ctx = t_kernel.make_launch_context()
-        launch_ctx_buffer: DefaultDict[KernelBatchedArgType, list[tuple]] = defaultdict(list)
-        callbacks: list[Callable[[], None]] = []
+        # Keep track of taichi runtime to automatically clear cache if destroyed
+        if self._prog_weakref is None:
+            prog = impl.get_runtime().prog
```

Comment:
can you explain why we don't simply systematiclly call `prog = impl.get_runtime().prog` ?